shared/native/notification-listeners.shared.js

```diff
@@ -4,64 +4,66 @@ import * as ConfigGen from '../actions/config-gen'
 import * as LoginGen from '../actions/login-gen'
 import * as NotificationsGen from '../actions/notifications-gen'
 import {throttle} from 'lodash-es'
-
-import type {Dispatch} from '../constants/types/flux'
-import type {IncomingCallMapType} from '../constants/types/rpc-gen'
+import engine from '../engine'
 
 // Keep track of the last time we notified and ignore if it's the same
 let lastLoggedInNotifyUsername = null
 
 // We get a counter for badge state, if we get one that's less than what we've seen we toss it
 let lastBadgeStateVersion = -1
 
-export default function(dispatch: Dispatch, getState: () => Object, notify: any): IncomingCallMapType {
-  const throttledDispatch = throttle(action => dispatch(action), 1000, {
-    leading: false,
-    trailing: true,
+// TODO: DESKTOP-6662 - Move notification listeners to their own actions
+export default (): void => {
+  // const throttledDispatch = throttle(action => dispatch(action), 1000, {
+  //   leading: false,
+  //   trailing: true,
+  // })
+
+  engine().setIncomingActionCreators('keybase.1.NotifyBadges.badgeState', ({badgeState}) => {
+    if (badgeState.inboxVers < lastBadgeStateVersion) {
+      logger.info(
+        `Ignoring older badgeState, got ${badgeState.inboxVers} but have seen ${lastBadgeStateVersion}`
+      )
+      return
+    }
+
+    lastBadgeStateVersion = badgeState.inboxVers
+
+    const conversations = badgeState.conversations
+    const totalChats = (conversations || []).reduce((total, c) => total + c.unreadMessages, 0)
+    const action = NotificationsGen.createReceivedBadgeState({badgeState})
+    if (totalChats > 0) {
+      // Defer this slightly so we don't get flashing if we're quickly receiving and reading
+      // throttledDispatch(action)
```

Comment:
I still don't know how to use `Saga.throttle` (which would need to be added to `util/saga`) to replace this.